File: python/structural/compositeObject/MainCompositeObject.py
import random
from abc import ABC, abstractmethod

class Element(ABC):
    @abstractmethod
    def render(self) -> str:
        pass

class Text(Element):
    def __init__(self, text: str):
        self.id = random.random()
        self.text = text

    def render(self) -> str:
        return self.text;

class View(Element):
    def __init__(self):
        self.id = random.random()
        self.elements: [Element] = list()

    def render(self) -> str:
        result: str = "";
        for element in self.elements:
            result += element.render() + "\n"
        return result

    def add(self, element: Element):
        self.elements.append(element)

    def remove(self, element: Element):
        self.elements.remove(element)

    def getChilds(self) -> [Element]:
        return self.elements

def main():
    text1 = Text("text1")
    text2 = Text("text2")
    view = View()
    view.add(text1)
    view.add(text2)
    view2 = View()
    text3 = Text("text3")
    view2.add(view)
    view2.add(text3)
    print(view2.render())

main()

File: kotlin/src/main/kotlin/structural/compositeObject/MainCompositeObject.kt
package structural.compositeObject

import kotlin.random.Random

internal interface Element {
    fun render(): String
}

internal class Text(private val text: String) : Element {
    private val id: Int = Random(1).nextInt()
    override fun render(): String {
        return text
    }

    override fun hashCode(): Int {
        return id
    }

    override fun equals(obj: Any?): Boolean {
        if (this === obj) return true
        if (obj == null) return false
        val other = obj as Text
        return id == other.id
    }
}

internal class View : Element {
    private val id: Int = Random(2).nextInt()
    private val elements: MutableList<Element> = ArrayList()
    override fun render(): String {
        var result = ""
        for (element in elements) {
            result += "${element.render()}\n"
        }
        return result
    }

    fun add(element: Element) {
        elements.add(element)
    }

    fun remove(element: Element) {
        elements.remove(element)
    }

    val childs: List<Element>
        get() = elements

    override fun hashCode(): Int {
        return id
    }

    override fun equals(obj: Any?): Boolean {
        if (this === obj) return true
        if (obj == null) return false
        val other = obj as View
        return id == other.id
    }
}

fun main(args: Array<String>) {
    val text1 = Text("text1")
    val text2 = Text("text2")
    val view = View()
    view.add(text1)
    view.add(text2)
    val view2 = View()
    val text3 = Text("text3")
    view2.add(view)
    view2.add(text3)
    println(view2.render())
}

